utilities/test_suite/HOST/Tensor_misc_host.cpp

```diff
@@ -0,0 +1,548 @@
+{
+    if (descriptorPtr->numDims > 0)
+    {
+        uint64_t v = 1;
+        for (int i = descriptorPtr->numDims - 1; i > 0; i--)
+        {
+            descriptorPtr->strides[i] = v;
+            v *= descriptorPtr->dims[i];
+        }
+        descriptorPtr->strides[0] = v;
+    }
+}
+
+string get_path(Rpp32u nDim, Rpp32u readType)
+{
+    string refPath = get_current_dir_name();
+    string pattern = "HOST/build";
+    string finalPath = "";
+    remove_substring(refPath, pattern);
+    string dim = std::to_string(nDim) + "d";
+
+    if (readType == 0 || readType == 2 || readType == 3)
+        finalPath = refPath + "NORMALIZE/input/" + dim;
+    else
+        finalPath = refPath + "NORMALIZE/output/" + dim;
+
+    return finalPath;
+}
+
+Rpp32u get_buffer_length(Rpp32u nDim, int axisMask)
+{
+    string dimSpecifier = std::to_string(nDim) + "d";
+    string refPath = get_path(nDim, 0);
+    string refFile;
+    if(axisMask != 6)
+        refFile = refPath + "/" + dimSpecifier + "_" + "input" + "_3x4x16.txt";
+    else
+        refFile = refPath + "/" + dimSpecifier + "_" + "input" + "_4x5x7.txt";
+    ifstream file(refFile);
+    Rpp32u bufferLength = std::count(std::istreambuf_iterator<char>(file), std::istreambuf_iterator<char>(), '\n');
+    return bufferLength;
+}
+
+void read_data(Rpp32f *data, Rpp32u nDim, Rpp32u readType, Rpp32u bufferLength, Rpp32u batchSize, Rpp32u axisMask)
+{
+    Rpp32u sampleLength = bufferLength / batchSize;
+    if(nDim != 3 && nDim != 4)
+    {
+        std::cout<<"\nGolden Inputs / Outputs are generated only for 3D/4D data"<<std::endl;
+        exit(0);
+    }
+
+    string refPath = get_path(nDim, readType);
+    string dimSpecifier = std::to_string(nDim) + "d";
+    string type = "input";
+    if (readType == 1)
+        type = "output";
+
+    string refFilePath;
+    if (readType == 1)
+        refFilePath = refPath + "/" + dimSpecifier + "_axisMask" + std::to_string(axisMask) + ".txt";
+    else
+    {
+        if(axisMask != 6)
+            refFilePath = refPath + "/" + dimSpecifier + "_" + type + "_3x4x16.txt";
+        else
+            refFilePath = refPath + "/" + dimSpecifier + "_" + type + "_4x5x7.txt";
+    }
+    fstream refFile;
+    refFile.open(refFilePath, ios::in);
+    if(!refFile.is_open())
+        cout<<"Unable to open the file specified! Please check the path of the file given as input"<<endl;
+
+    for(int i = 0; i < batchSize; i++)
+    {
+        Rpp32f *curData = data + i * sampleLength;
+        for(int j = 0; j < sampleLength; j++)
+            refFile >> curData[j];
+    }
+}
+
+// Fill the starting indices and length of ROI values
+void fill_roi_values(Rpp32u nDim, Rpp32u batchSize, Rpp32u *roiTensor, bool qaMode)
+{
+    switch(nDim)
+    {
+        case 3:
+        {
+            if (qaMode)
+            {
+                for(int i = 0; i < batchSize * 6; i += 6)
+                {
+                    roiTensor[i] = 0;
```

Comment:
Possibly just initialize in one line with std::array<Rpp32u, 6> and do a std::copy of a pointer to that array batchSize times